packages/react/src/host/RemoteComponent.tsx

```diff
@@ -46,6 +47,7 @@ export const RemoteComponent = memo(
               receiver={receiver}
               component={child}
               controller={controller}
+              __type__={(controller.get(child.type) as any)?.__type__}
```

Comment:
`__type__` is somewhat vague. The code in polaris-react isn't clear that this is due to argo-related functionality. I worry that in 3 month when people come to look at the code in polaris-react they won't have a clue what the extra logic is for and thus might be deleted.

Could this be named more explicitly? A longer name more explicit name would also reduce the potential for accidental naming clashes and thus accidental triggering of the logic in polaris-react. `__remoteUiChildType__` perhaps?

Reply:
Same as https://github.com/Shopify/polaris-react/pull/3517#discussion_r508042115